tasks: Add translation tasks.

trueskawka opened the pull request:
Initial proposal for GCI translation tasks - requires refinement, a list of languages we want to create tasks for and upload script.

@timabbott @rishig could you take a look?

smarx commented:
*Automated message from Dropbox CLA bot*

@trueskawka, it looks like you've already signed the Dropbox CLA. Thanks!

trueskawka commented:
Added the upload script. Proposed list of languages:

* Simplified Chinese
* German
* Russian
* Italian
* French
* Turkish
* Portuguese
* Arabic 
* Hindi

Am I not to Eurocentric?

rishig commented on scripts/translations.py:
```diff
@@ -0,0 +1,98 @@
+args = parser.parse_args()
+
+languages = [
+    ('Simplified Chinese'),
```
We can also add Japanese, Malayalam, Serbian, Hungarian, and Korean (all have at least 50 strings translated on transifex).

rishig commented on scripts/translations.py:
```diff
@@ -0,0 +1,98 @@
+    ('Portuguese'),
+    ('Arabic'),
+    ('Hindi'),
+]
```
It would also be great to add a task that allows the student to propose a language on chat.zulip.org, and if we approve it, they can do it (with the idea that we'd approve basically anything).

rishig commented on scripts/translations.py:
```diff
@@ -0,0 +1,98 @@
+    #Task Type A
+    upload_task(
+        # https://developers.google.com/open-source/gci/resources/downloads/TaskAPISpec.pdf
+        name = 'Research web application translations into %s' % (language),
```
A lot of the tasks I wrote don't do this, but I would put the %s earlier in the name, so that it doesn't get cut-off by the narrow "name" column in google's task display. So something like "Research %s web application translations."

rishig commented on tasks/translations.md:
```diff
@@ -0,0 +1,132 @@
+# GCI Tasks: Creating translation style guides
+
+## Prerequisites
+
```
You can copy this from bot-avatars:

* You need to know how to create a GitHub pull request. Check out the
  [Learn how to create a GitHub Pull Request](https://codein.withgoogle.com/tasks/6541581402243072/)
  task if you aren't sure how to do this, or read through the task description
  [here](https://github.com/zulip/zulip-gci/blob/master/tasks/submit-a-pull-request.md).

rishig commented on tasks/translations.md:
```diff
@@ -0,0 +1,132 @@
+# GCI Tasks: Creating translation style guides
+
+## Prerequisites
```
I would also add
* Native fluency in the appropriate language.

as a prerequisite

rishig commented on tasks/translations.md:
```diff
@@ -0,0 +1,132 @@
+
+### Task Type B: Write general rules for translations
+
+* Create a new file `general_rules.md` in `translations/<language>`.
```
I think we've been using general-rules.md rather than general_rules.md as our convention for documentation files.

rishig commented on tasks/translations_terms.md:
```diff
@@ -0,0 +1,21 @@
+* **filter**: one of the options for viewing different kinds of messages, listed in the upper left corner of the window: Home, Private messages, Starred messages, @-mentions. Also called a "view".
+* **group PM**: a private message sent directly to two or more other users rather than through a stream.
+* **home**: a "view" in which all topics in all subscribed streams are visible, in order by date and time of posting
+* **me**: a reference to one's own name, formatted as `/me`.
```
I guess in theory we could support this being translated, though for now maybe we should mention that it shouldn't be?

rishig commented on tasks/translations.md:
```diff
@@ -0,0 +1,132 @@
+* Write rules for translations specific to the *language*. If applicable, answer
+the following questions:
+
+  * should the language be formal or informal?
```
These bullets can all be capitalized.

rishig commented:
This is awesome, thanks for working on this @trueskawka! Made some minor comments.

trueskawka commented:
Thanks, @rishig, I've implemented your comments! I've also improved the tasks after a chat with @timabbott.

Major changes:
- added Task Type D for improving an existing style guide
- added Task Type E for proposing a language
- refactored the list of terms and task descriptions to include the [Chinese style guide](https://zulip.readthedocs.io/en/latest/chinese.html) approach
- refactored the upload script to include tasks for refactoring Polish and Spanish guides, and proposing languages
- changed time to complete to 5 days, since students tend to rush through tasks

rishig commented:
Tasks uploaded, thanks @trueskawka!
